src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -639,10 +640,22 @@ public Repository getExistingCheckedOutRepositoryByLocalPath(@NotNull Path local
      */
     public void commit(Repository repo, String message) throws GitAPIException {
         try (Git git = new Git(repo)) {
-            git.commit().setMessage(message).setAllowEmpty(true).setCommitter(artemisGitName, artemisGitEmail).call();
+            commit(git).setMessage(message).setAllowEmpty(true).setCommitter(artemisGitName, artemisGitEmail).call();
         }
     }
 
+    /**
+     * Creates a CommitCommand and sets signing to false. Egit uses the local git configuration and if signing of
+     * commits is enabled, tests will fail because it will not be able to actually sign the commit.
+     * This method makes sure that signing is disabled and commits work on systems regardless of the local git configuration.
+     *
+     * @param git Git Repository Object.
+     * @return CommitCommand with signing set to false.
+     */
+    public CommitCommand commit(Git git) {
```

Comment:
Could this method just be static? If yes, then you would not need to add the service as dependency to new locations.

Reply:
It can yes, I changed it. Thanks!